src/PQueue.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+export default class PQueue {
+
+  constructor() {
+    this._list = []
+    this._backElementIndex = -1
+  }
+
+  enqueue( element, priority ) {
+    if ( priority < 1 ) {
+      throw new Error( 'This priority is invalid' )
+    }
+    this._list[ ++this._backElementIndex ] = { element, priority }
```

Comment:
Consider the use of `Node` objects to encapsulate the data and behavior of entries in the `PriorityQueue`.